### ISSUE TITLE
Rustdoc: long lines are escaping text code blocks 

### ISSUE THREAD

**Issue by WaffleLapkin** [rust-lang/rust#87043]
Repro: 

```rust
/// ```text
/// aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
/// ```
/// ```rust
/// bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb
/// ```
pub struct Item;
```
Rustdoc output:
![image](https://user-images.githubusercontent.com/38225716/125177499-f5f1f500-e1e4-11eb-9a77-03ab867f8ee8.png)

It seems that only code block with `text` language are affected.

cc @GuillaumeGomez

### Meta

```text
; rustdoc --version --verbose
rustdoc 1.55.0-nightly (240ff4c4a 2021-07-09)
binary: rustdoc
commit-hash: 240ff4c4a0d0936c9eeb783fa9ff5c0507a6ffb4
commit-date: 2021-07-09
host: x86_64-unknown-linux-gnu
release: 1.55.0-nightly
LLVM version: 12.0.1
```

**Comment by WaffleLapkin** [comment]
A quick test showed that the bug was introduced in `nightly-2021-06-30` (i.e.: `nightly-2021-06-29` works fine)

**Comment by GuillaumeGomez** [comment]
Thanks, I'll fix it in the next hours.

**Comment by GuillaumeGomez** [comment]
In case you want to take a look to the fix, it's #87056.
